src/bitmap.cpp

```diff
@@ -876,6 +891,22 @@ void Bitmap::ClearRect(Rect const& dst_rect) {
 	RefreshCallback();
 }
 
+// Hard light lookup table mapping source color to destination color
+static int hard_light_lookup[256][256];
```

Comment:
I'm not sure if we would want this, but this could use constexpr to create the table at compile time like this:
```
struct hard_light_table {
	uint8_t table[256][256];
	constexpr hard_light_table() : table() {
	    for (int i = 0; i < 256; ++i) {
			for (int j = 0; j < 256; ++j) {
				int res = 0;
				if (i <= 128)
					res = (2 * i * j) / 255;
				else
					res = 255 - 2 * (255 - i) * (255 - j) / 255;
					table[i][j] = res > 255 ? 255 : res < 0 ? 0 : res;
			}
		}
	}
};
constexpr hard_light_table hard_light_lookup;
```
And replace hard_light_lookup with hard_light_lookup.table.